state/factory/factory.go

```diff
@@ -428,7 +431,7 @@ func (sf *factory) SimulateExecution(
 		return nil, nil, errors.Wrap(err, "failed to obtain working set from state factory")
 	}
 
-	return simulateExecution(ctx, ws, caller, ex, getBlockHash)
+	return simulateExecution(ctx, ws, caller, ex, getBlockHash, depositGas)
```

Comment:
because it is a simulation, we can put a dummy depositGas function here.

Reply:
yes! address in latest commit